datalab/mlalpha/commands/_ml.py

```diff
@@ -158,3 +181,57 @@ def _predict(args, cell):
 
 def _batch_predict(args, cell):
   return _run_package(args, cell, 'batch_predict')
+
+
+def _plot_confusion_matrix(cm, labels):
+  plt.imshow(cm, interpolation='nearest', cmap=plt.cm.Blues)
+  plt.title('Confusion matrix')
+  plt.colorbar()
+  tick_marks = np.arange(len(labels))
+  plt.xticks(tick_marks, labels, rotation=45)
+  plt.yticks(tick_marks, labels)
+  plt.tight_layout()
+  plt.ylabel('True label')
+  plt.xlabel('Predicted label')
+
+
+def _confusion_matrix(args, cell):
+  if args['csv'] is not None:
+    input_csv = args['csv']
+    schema_file = input_csv + '.schema.yaml'
+    headers = None
+    if cell is not None:
+      env = datalab.utils.commands.notebook_environment()
+      config = datalab.utils.commands.parse_config(cell, env)
+      headers_str = config.get('headers', None)
+      if headers_str is not None:
+        headers = [x.strip() for x in headers_str.split(',')]
+    if headers is not None:
+      with cloudml.util._file.open_local_or_gcs(input_csv, mode='r') as f:
+        df = pd.read_csv(f, names=headers)
+    elif cloudml.util._file.file_exists(schema_file):
+      df = datalab.mlalpha.csv_to_dataframe(input_csv, schema_file)
+    else:
+      raise Exception('headers is missing from cell, ' +
+                      'and there is no schema file in the same dir as csv')
+    labels=sorted(set(df['target']))
+    cm = confusion_matrix(df['target'], df['predicted'], labels=labels)
+    _plot_confusion_matrix(cm, labels)
+  elif args['bqtable'] is not None or args['sql'] is not None:
+    if args['sql'] is not None:
+      item = datalab.utils.commands.get_notebook_item(args['sql'])
+      query, _ = datalab.data.SqlModule.get_sql_statement_with_environment(item, {})
+    else:
+      query = ('select target, predicted, count(*) as count from %s group by target, predicted'
+               % args['bqtable'])
+    dfbq = datalab.bigquery.Query(query).results().to_dataframe()
+    labels = sorted(set(dfbq['target']))
```

Comment:
This assumes the labels for the target and predicted column are all the same. This might not be true; for example, what if I run prediction on a test set that is missing a class label. 

I think you want something like "labels = sorted(set(dfbq['target'])+set(dfbq['predicted']))" This still could miss labels if they never show up in the evaluation set or prediction set, but I guess this is ok (but still add a comment)